TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/RootOtelContext.java

```diff
@@ -0,0 +1,139 @@
+package org.opensearch.migrations.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.exporter.otlp.logs.OtlpGrpcLogRecordExporter;
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.logs.SdkLoggerProvider;
+import io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.opensearch.migrations.Utils;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+public class RootOtelContext implements IRootOtelContext {
+    private final OpenTelemetry openTelemetryImpl;
+    private final String scopeName;
+    @Getter
+    @Setter
+    Exception observedExceptionToIncludeInMetrics;
+
+    public static OpenTelemetry initializeOpenTelemetryForCollector(@NonNull String collectorEndpoint,
+                                                                    @NonNull String serviceName) {
+        var serviceResource = Resource.getDefault().toBuilder()
+                .put(ResourceAttributes.SERVICE_NAME, serviceName)
+                .build();
+
+        final var spanProcessor = BatchSpanProcessor.builder(OtlpGrpcSpanExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        .setTimeout(2, TimeUnit.SECONDS)
+                        .build())
+                .setScheduleDelay(100, TimeUnit.MILLISECONDS)
+                .build();
+        final var metricReader = PeriodicMetricReader.builder(OtlpGrpcMetricExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        // see https://opentelemetry.io/docs/specs/otel/metrics/sdk_exporters/prometheus/
+                        // "A Prometheus Exporter MUST only support Cumulative Temporality."
+                        //.setAggregationTemporalitySelector(AggregationTemporalitySelector.deltaPreferred())
```

Comment:
Instead of having this code commented which would break the Prometheus exporter, should we instead call

```java
.setAggregationTemporalitySelector(AggregationTemporalitySelector.alwaysCumulative())
```

even thought it is the default in the builder to make it explicit as a requirement in our usage.

Reply:
Do you think it would make sense to do both?  I'd like to be able to find the calls when I'm searching and looking at our configuration code.  These APIs are fast evolving so maybe it changes and, more likely, there's a good chance that somebody not very familiar with otel will try to make the same mistake.  The goal here isn't to make sure that we do what prometheus needs (we may not always use prometheus) but to give the reason that we can't do more.